## Estimating the Sharpe Ratio in Python

*Suggested Answers follow (usually there are multiple ways to solve a problem in Python).*

Obtain data for Microsoft and S&P 500 for the period 1st of January 2012 – 31st of December 2016 from Yahoo Finance.
Let S&P 500 act as the market.
Calculate the beta of Microsoft. <br />

Assume a risk-free rate of 2.5% and a risk premium of 5%.<br />
Estimate the expected return of Microsoft.



In [4]:


import numpy as np
import pandas as pd
import yfinance as yf
from pandas_datareader import data as wb

tickers = ['MSFT', '^GSPC']
data = pd.DataFrame()

for a in tickers:
    data[a] = yf.download(a, start='2012-1-1', end='2016-12-31', auto_adjust=False)['Adj Close']


sec_returns = np.log( data / data.shift(1) )
cov = sec_returns.cov() * 250
cov_with_market = cov.iloc[0,1]
market_var = sec_returns['^GSPC'].var() * 250

MSFT_beta = cov_with_market / market_var


sec_returns = np.log( data / data.shift(1) )
cov = sec_returns.cov() * 250
cov_with_market = cov.iloc[0,1]
market_var = sec_returns['^GSPC'].var() * 250

MSFT_beta = cov_with_market / market_var
MSFT_er = 0.025 + MSFT_beta * 0.05

[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed


Calculate the Sharpe ratio in Python.

Chỉ số Sharpe là một thước đo cực kỳ quan trọng, dùng để đo lường tỷ suất sinh lợi đã điều chỉnh rủi ro (risk-adjusted return).

**Sharpe ratio:**
### $$
Sharpe = \frac{\overline{r_{pg}} - r_f}{\sigma_{pg}}
$$

In [5]:
Sharpe = (MSFT_er - 0.025) / (sec_returns['MSFT'].std() * 250 ** 0.5)
Sharpe

np.float64(0.23999691281001811)

1. Ý nghĩa của Công thức
Chỉ số Sharpe trả lời câu hỏi: "Tôi nhận được bao nhiêu lợi nhuận vượt trội (trên mức phi rủi ro) cho mỗi đơn vị tổng rủi ro mà tôi chấp nhận?"

Không giống như Beta (chỉ đo rủi ro hệ thống), chỉ số Sharpe sử dụng độ lệch chuẩn (tổng rủi ro) ở mẫu số. Điều này làm cho nó trở thành một thước đo hiệu quả toàn diện hơn, vì nó cho biết liệu lợi nhuận của một tài sản có xứng đáng với toàn bộ sự biến động (rủi ro) của nó hay không.

2. Ý nghĩa của Kết quả (Sharpe = 0.24)
Kết quả 0.24 có ý nghĩa so sánh:

Sharpe < 1: Thường được coi là một khoản đầu tư không hiệu quả. Lợi nhuận vượt trội bạn nhận được (tử số) thấp hơn mức rủi ro (mẫu số) bạn phải gánh chịu.

Sharpe > 1: Được coi là một khoản đầu tư tốt, vì lợi nhuận vượt trội cao hơn rủi ro gánh chịu.

Sharpe > 2: Được coi là rất tốt.

Với kết quả 0.24, chúng ta có thể kết luận rằng trong giai đoạn 2012-2016 và với các giả định đã cho:

Khoản đầu tư vào MSFT không mang lại hiệu quả cao khi xét trên góc độ rủi ro/lợi nhuận.

Nhà đầu tư nhận được chỉ 0.24 đơn vị lợi nhuận vượt trội cho mỗi 1 đơn vị rủi ro mà họ gánh chịu. Nói cách khác, sự biến động (rủi ro) của MSFT là tương đối lớn so với phần lợi nhuận vượt trội mà nó tạo ra.